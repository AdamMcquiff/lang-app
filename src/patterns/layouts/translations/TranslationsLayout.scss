.l-translations {
    background-color: $dark-blue;
}