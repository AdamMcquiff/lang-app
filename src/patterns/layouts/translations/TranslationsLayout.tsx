import { observer } from 'mobx-react';
import * as React from 'react';
import ITranslationCategoryModel from '../../../models/TranslationCategory';
import RootStore from '../../../store/RootStore';
import CategoryRow from '../../organisms/categoryRow/CategoryRow';
import './TranslationsLayout.scss';

interface ITranslationInterface {
    store: RootStore,
    className: string
}

@observer
class TranslationsLayout extends React.Component<ITranslationInterface> {
    public render() {
        return (
            <main className={'l-translations ' + this.props.className}>
                {this.props.store.translationCategoryStore.translationCategories.map((category: ITranslationCategoryModel, id: number) => {
                    return <CategoryRow key={id} 
                                        category={category} 
                                        settingsStore={this.props.store.settingsStore}
                                        translationStore={this.props.store.translationStore} />
                })}
            </main>
        )
    }
}

export default TranslationsLayout;