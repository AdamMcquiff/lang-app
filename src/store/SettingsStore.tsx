import { observable } from 'mobx';
import { ISettingsModel } from '../models/Settings';
import { ISettingsService } from '../services/settingsService/SettingsServiceInterface';
import { RootStore } from './RootStore';

export class SettingsStore {
    @observable 
    public settings: ISettingsModel = {
        native_lang: '',
        translated_lang: ''
    };

    public rootStore: RootStore;
    public service: ISettingsService;
    
    constructor(rootStore: RootStore, service: ISettingsService) {
        this.rootStore = rootStore;
        this.service = service;
    }

    public list() {
        this.service.list().then((settings: ISettingsModel) => {
            this.settings = settings;
        });
    }
}