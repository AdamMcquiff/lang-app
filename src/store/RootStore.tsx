import SettingsService from '../services/settingsService/SettingsService';
import TranslationCategoryService from '../services/translationCategoriesService/TranslationCategoryService';
import TranslationService from '../services/translationService/TranslationService';
import SettingsStore from './SettingsStore';
import TranslationCategoryStore from './TranslationCategoryStore';
import TranslationStore from './TranslationStore';

class RootStore {
	public settingsStore: SettingsStore;
	public translationCategoryStore: TranslationCategoryStore;
	public translationStore: TranslationStore;

	constructor() {
		this.settingsStore = new SettingsStore(this, SettingsService);
		this.translationCategoryStore = new TranslationCategoryStore(this, TranslationCategoryService);
		this.translationStore = new TranslationStore(this, TranslationService);
	}
}

export default RootStore;