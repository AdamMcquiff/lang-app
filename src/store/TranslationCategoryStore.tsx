import { observable } from 'mobx';
import { ITranslationCategoryModel } from '../models/TranslationCategory';
import { ITranslationCategoryService } from '../services/translationCategoriesService/TranslationCategoryServiceInterface';
import { RootStore } from './RootStore';

export class TranslationCategoryStore {
    @observable 
    public translationCategories: ITranslationCategoryModel[] = [];

    public rootStore: RootStore;
    public service: ITranslationCategoryService;

    constructor(rootStore: RootStore, service: ITranslationCategoryService) {
        this.rootStore = rootStore
        this.service = service;
    }

    public list = () => {
        this.service.list().then((categories: ITranslationCategoryModel[]) => {
            this.translationCategories = categories;
        });
    }

    public create = (category: ITranslationCategoryModel) => {
        this.service.create(category).then((data: ITranslationCategoryModel) => {
            this.translationCategories.push(data);
        });
    }

    public delete = (category: ITranslationCategoryModel) => {
        this.service.delete(category).then(() => {
            this.translationCategories = this.translationCategories.filter(
                item => item.id !== category.id ? item : false
            );
        });
    }
}