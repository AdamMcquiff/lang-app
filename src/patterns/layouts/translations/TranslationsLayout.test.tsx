import { shallow } from 'enzyme';
import * as React from 'react';
import { RootStore } from '../../../store/RootStore';
import { TranslationsLayout } from './TranslationsLayout';

const store = new RootStore();
store.translationCategoryStore.translationCategories = [{
    id: 1, title: 'Some category'
}]

describe('TranslationLayout', () => {
    const props = { 
        className: 'some class',
        store 
    };

    it('renders without crashing with all props', () => {
        const component = shallow(<TranslationsLayout {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('renders without crashing with only required props', () => {
        const component = shallow(<TranslationsLayout store={props.store} />);
        expect(component).toMatchSnapshot();
    });

    it('renders correct number of CategoryRow elements', () => {
        const component = shallow(<TranslationsLayout store={props.store} />);
        expect(component.find('CategoryRow').length).toEqual(1);
    });
});