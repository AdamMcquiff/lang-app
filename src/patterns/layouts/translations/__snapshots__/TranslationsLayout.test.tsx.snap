// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationLayout renders without crashing with all props 1`] = `
<main
  className="l-translations some class"
>
  <CategoryRow
    category={
      Object {
        "id": 1,
        "title": "Some category",
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "TranslationCategoryStore@2.translationCategories[..].keys",
            "observers": Set {},
          },
          "name": "TranslationCategoryStore@2.translationCategories[..]",
          "pendingKeys": Map {
            "cheerio" => false,
            "$$typeof" => false,
            "nodeType" => false,
            "@@__IMMUTABLE_ITERABLE__@@" => false,
            "@@__IMMUTABLE_RECORD__@@" => false,
            "_isMockFunction" => false,
            "toJSON" => false,
          },
          "proxy": [Circular],
          "target": Object {
            "id": 1,
            "title": "Some category",
            Symbol(mobx administration): [Circular],
          },
          "values": Map {
            "id" => 1,
            "title" => "Some category",
          },
        },
      }
    }
    key="0"
    settingsStore={
      SettingsStore {
        "rootStore": RootStore {
          "settingsStore": [Circular],
          "translationCategoryStore": TranslationCategoryStore {
            "create": [Function],
            "list": [Function],
            "rootStore": [Circular],
            "service": TranslationCategoryService {},
            "translationCategories": Array [
              Object {
                "id": 1,
                "title": "Some category",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                    "observers": Set {},
                  },
                  "name": "TranslationCategoryStore@2.translationCategories[..]",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "id" => 1,
                    "title" => "Some category",
                  },
                },
              },
            ],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "TranslationCategoryStore@2.keys",
                "observers": Set {},
              },
              "name": "TranslationCategoryStore@2",
              "target": [Circular],
              "values": Map {
                "translationCategories" => Array [
                  Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): ObservableObjectAdministration$$1 {
                      "defaultEnhancer": [Function],
                      "keysAtom": Atom$$1 {
                        "diffValue": 0,
                        "isBeingObserved": false,
                        "isPendingUnobservation": false,
                        "lastAccessedBy": 0,
                        "lowestObserverState": 2,
                        "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                        "observers": Set {},
                      },
                      "name": "TranslationCategoryStore@2.translationCategories[..]",
                      "pendingKeys": Map {
                        "cheerio" => false,
                        "$$typeof" => false,
                        "nodeType" => false,
                        "@@__IMMUTABLE_ITERABLE__@@" => false,
                        "@@__IMMUTABLE_RECORD__@@" => false,
                        "_isMockFunction" => false,
                        "toJSON" => false,
                      },
                      "proxy": [Circular],
                      "target": Object {
                        "id": 1,
                        "title": "Some category",
                        Symbol(mobx administration): [Circular],
                      },
                      "values": Map {
                        "id" => 1,
                        "title" => "Some category",
                      },
                    },
                  },
                ],
              },
            },
          },
          "translationStore": TranslationStore {
            "create": [Function],
            "delete": [Function],
            "list": [Function],
            "rootStore": [Circular],
            "service": TranslationService {},
            "translations": Array [],
            "update": [Function],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "TranslationStore@3.keys",
                "observers": Set {},
              },
              "name": "TranslationStore@3",
              "target": [Circular],
              "values": Map {
                "translations" => Array [],
              },
            },
          },
        },
        "service": SettingsService {},
        "settings": Object {
          "native_lang": "",
          "translated_lang": "",
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "SettingsStore@1.settings.keys",
              "observers": Set {},
            },
            "name": "SettingsStore@1.settings",
            "pendingKeys": Map {
              "cheerio" => false,
              "$$typeof" => false,
              "nodeType" => false,
              "@@__IMMUTABLE_ITERABLE__@@" => false,
              "@@__IMMUTABLE_RECORD__@@" => false,
              "_isMockFunction" => false,
              "toJSON" => false,
            },
            "proxy": [Circular],
            "target": Object {
              "native_lang": "",
              "translated_lang": "",
              Symbol(mobx administration): [Circular],
            },
            "values": Map {
              "native_lang" => "",
              "translated_lang" => "",
            },
          },
        },
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "SettingsStore@1.keys",
            "observers": Set {},
          },
          "name": "SettingsStore@1",
          "target": [Circular],
          "values": Map {
            "settings" => Object {
              "native_lang": "",
              "translated_lang": "",
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "SettingsStore@1.settings.keys",
                  "observers": Set {},
                },
                "name": "SettingsStore@1.settings",
                "pendingKeys": Map {
                  "cheerio" => false,
                  "$$typeof" => false,
                  "nodeType" => false,
                  "@@__IMMUTABLE_ITERABLE__@@" => false,
                  "@@__IMMUTABLE_RECORD__@@" => false,
                  "_isMockFunction" => false,
                  "toJSON" => false,
                },
                "proxy": [Circular],
                "target": Object {
                  "native_lang": "",
                  "translated_lang": "",
                  Symbol(mobx administration): [Circular],
                },
                "values": Map {
                  "native_lang" => "",
                  "translated_lang" => "",
                },
              },
            },
          },
        },
      }
    }
    translationStore={
      TranslationStore {
        "create": [Function],
        "delete": [Function],
        "list": [Function],
        "rootStore": RootStore {
          "settingsStore": SettingsStore {
            "rootStore": [Circular],
            "service": SettingsService {},
            "settings": Object {
              "native_lang": "",
              "translated_lang": "",
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "SettingsStore@1.settings.keys",
                  "observers": Set {},
                },
                "name": "SettingsStore@1.settings",
                "pendingKeys": Map {
                  "cheerio" => false,
                  "$$typeof" => false,
                  "nodeType" => false,
                  "@@__IMMUTABLE_ITERABLE__@@" => false,
                  "@@__IMMUTABLE_RECORD__@@" => false,
                  "_isMockFunction" => false,
                  "toJSON" => false,
                },
                "proxy": [Circular],
                "target": Object {
                  "native_lang": "",
                  "translated_lang": "",
                  Symbol(mobx administration): [Circular],
                },
                "values": Map {
                  "native_lang" => "",
                  "translated_lang" => "",
                },
              },
            },
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "SettingsStore@1.keys",
                "observers": Set {},
              },
              "name": "SettingsStore@1",
              "target": [Circular],
              "values": Map {
                "settings" => Object {
                  "native_lang": "",
                  "translated_lang": "",
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "SettingsStore@1.settings.keys",
                      "observers": Set {},
                    },
                    "name": "SettingsStore@1.settings",
                    "pendingKeys": Map {
                      "cheerio" => false,
                      "$$typeof" => false,
                      "nodeType" => false,
                      "@@__IMMUTABLE_ITERABLE__@@" => false,
                      "@@__IMMUTABLE_RECORD__@@" => false,
                      "_isMockFunction" => false,
                      "toJSON" => false,
                    },
                    "proxy": [Circular],
                    "target": Object {
                      "native_lang": "",
                      "translated_lang": "",
                      Symbol(mobx administration): [Circular],
                    },
                    "values": Map {
                      "native_lang" => "",
                      "translated_lang" => "",
                    },
                  },
                },
              },
            },
          },
          "translationCategoryStore": TranslationCategoryStore {
            "create": [Function],
            "list": [Function],
            "rootStore": [Circular],
            "service": TranslationCategoryService {},
            "translationCategories": Array [
              Object {
                "id": 1,
                "title": "Some category",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                    "observers": Set {},
                  },
                  "name": "TranslationCategoryStore@2.translationCategories[..]",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "id" => 1,
                    "title" => "Some category",
                  },
                },
              },
            ],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "TranslationCategoryStore@2.keys",
                "observers": Set {},
              },
              "name": "TranslationCategoryStore@2",
              "target": [Circular],
              "values": Map {
                "translationCategories" => Array [
                  Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): ObservableObjectAdministration$$1 {
                      "defaultEnhancer": [Function],
                      "keysAtom": Atom$$1 {
                        "diffValue": 0,
                        "isBeingObserved": false,
                        "isPendingUnobservation": false,
                        "lastAccessedBy": 0,
                        "lowestObserverState": 2,
                        "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                        "observers": Set {},
                      },
                      "name": "TranslationCategoryStore@2.translationCategories[..]",
                      "pendingKeys": Map {
                        "cheerio" => false,
                        "$$typeof" => false,
                        "nodeType" => false,
                        "@@__IMMUTABLE_ITERABLE__@@" => false,
                        "@@__IMMUTABLE_RECORD__@@" => false,
                        "_isMockFunction" => false,
                        "toJSON" => false,
                      },
                      "proxy": [Circular],
                      "target": Object {
                        "id": 1,
                        "title": "Some category",
                        Symbol(mobx administration): [Circular],
                      },
                      "values": Map {
                        "id" => 1,
                        "title" => "Some category",
                      },
                    },
                  },
                ],
              },
            },
          },
          "translationStore": [Circular],
        },
        "service": TranslationService {},
        "translations": Array [],
        "update": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "TranslationStore@3.keys",
            "observers": Set {},
          },
          "name": "TranslationStore@3",
          "target": [Circular],
          "values": Map {
            "translations" => Array [],
          },
        },
      }
    }
  />
</main>
`;

exports[`TranslationLayout renders without crashing with only required props 1`] = `
<main
  className="l-translations undefined"
>
  <CategoryRow
    category={
      Object {
        "id": 1,
        "title": "Some category",
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "TranslationCategoryStore@2.translationCategories[..].keys",
            "observers": Set {},
          },
          "name": "TranslationCategoryStore@2.translationCategories[..]",
          "pendingKeys": Map {
            "cheerio" => false,
            "$$typeof" => false,
            "nodeType" => false,
            "@@__IMMUTABLE_ITERABLE__@@" => false,
            "@@__IMMUTABLE_RECORD__@@" => false,
            "_isMockFunction" => false,
            "toJSON" => false,
          },
          "proxy": [Circular],
          "target": Object {
            "id": 1,
            "title": "Some category",
            Symbol(mobx administration): [Circular],
          },
          "values": Map {
            "id" => 1,
            "title" => "Some category",
          },
        },
      }
    }
    key="0"
    settingsStore={
      SettingsStore {
        "rootStore": RootStore {
          "settingsStore": [Circular],
          "translationCategoryStore": TranslationCategoryStore {
            "create": [Function],
            "list": [Function],
            "rootStore": [Circular],
            "service": TranslationCategoryService {},
            "translationCategories": Array [
              Object {
                "id": 1,
                "title": "Some category",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                    "observers": Set {},
                  },
                  "name": "TranslationCategoryStore@2.translationCategories[..]",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "id" => 1,
                    "title" => "Some category",
                  },
                },
              },
            ],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "TranslationCategoryStore@2.keys",
                "observers": Set {},
              },
              "name": "TranslationCategoryStore@2",
              "target": [Circular],
              "values": Map {
                "translationCategories" => Array [
                  Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): ObservableObjectAdministration$$1 {
                      "defaultEnhancer": [Function],
                      "keysAtom": Atom$$1 {
                        "diffValue": 0,
                        "isBeingObserved": false,
                        "isPendingUnobservation": false,
                        "lastAccessedBy": 0,
                        "lowestObserverState": 2,
                        "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                        "observers": Set {},
                      },
                      "name": "TranslationCategoryStore@2.translationCategories[..]",
                      "pendingKeys": Map {
                        "cheerio" => false,
                        "$$typeof" => false,
                        "nodeType" => false,
                        "@@__IMMUTABLE_ITERABLE__@@" => false,
                        "@@__IMMUTABLE_RECORD__@@" => false,
                        "_isMockFunction" => false,
                        "toJSON" => false,
                      },
                      "proxy": [Circular],
                      "target": Object {
                        "id": 1,
                        "title": "Some category",
                        Symbol(mobx administration): [Circular],
                      },
                      "values": Map {
                        "id" => 1,
                        "title" => "Some category",
                      },
                    },
                  },
                ],
              },
            },
          },
          "translationStore": TranslationStore {
            "create": [Function],
            "delete": [Function],
            "list": [Function],
            "rootStore": [Circular],
            "service": TranslationService {},
            "translations": Array [],
            "update": [Function],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "TranslationStore@3.keys",
                "observers": Set {},
              },
              "name": "TranslationStore@3",
              "target": [Circular],
              "values": Map {
                "translations" => Array [],
              },
            },
          },
        },
        "service": SettingsService {},
        "settings": Object {
          "native_lang": "",
          "translated_lang": "",
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "SettingsStore@1.settings.keys",
              "observers": Set {},
            },
            "name": "SettingsStore@1.settings",
            "pendingKeys": Map {
              "cheerio" => false,
              "$$typeof" => false,
              "nodeType" => false,
              "@@__IMMUTABLE_ITERABLE__@@" => false,
              "@@__IMMUTABLE_RECORD__@@" => false,
              "_isMockFunction" => false,
              "toJSON" => false,
            },
            "proxy": [Circular],
            "target": Object {
              "native_lang": "",
              "translated_lang": "",
              Symbol(mobx administration): [Circular],
            },
            "values": Map {
              "native_lang" => "",
              "translated_lang" => "",
            },
          },
        },
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "SettingsStore@1.keys",
            "observers": Set {},
          },
          "name": "SettingsStore@1",
          "target": [Circular],
          "values": Map {
            "settings" => Object {
              "native_lang": "",
              "translated_lang": "",
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "SettingsStore@1.settings.keys",
                  "observers": Set {},
                },
                "name": "SettingsStore@1.settings",
                "pendingKeys": Map {
                  "cheerio" => false,
                  "$$typeof" => false,
                  "nodeType" => false,
                  "@@__IMMUTABLE_ITERABLE__@@" => false,
                  "@@__IMMUTABLE_RECORD__@@" => false,
                  "_isMockFunction" => false,
                  "toJSON" => false,
                },
                "proxy": [Circular],
                "target": Object {
                  "native_lang": "",
                  "translated_lang": "",
                  Symbol(mobx administration): [Circular],
                },
                "values": Map {
                  "native_lang" => "",
                  "translated_lang" => "",
                },
              },
            },
          },
        },
      }
    }
    translationStore={
      TranslationStore {
        "create": [Function],
        "delete": [Function],
        "list": [Function],
        "rootStore": RootStore {
          "settingsStore": SettingsStore {
            "rootStore": [Circular],
            "service": SettingsService {},
            "settings": Object {
              "native_lang": "",
              "translated_lang": "",
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "SettingsStore@1.settings.keys",
                  "observers": Set {},
                },
                "name": "SettingsStore@1.settings",
                "pendingKeys": Map {
                  "cheerio" => false,
                  "$$typeof" => false,
                  "nodeType" => false,
                  "@@__IMMUTABLE_ITERABLE__@@" => false,
                  "@@__IMMUTABLE_RECORD__@@" => false,
                  "_isMockFunction" => false,
                  "toJSON" => false,
                },
                "proxy": [Circular],
                "target": Object {
                  "native_lang": "",
                  "translated_lang": "",
                  Symbol(mobx administration): [Circular],
                },
                "values": Map {
                  "native_lang" => "",
                  "translated_lang" => "",
                },
              },
            },
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "SettingsStore@1.keys",
                "observers": Set {},
              },
              "name": "SettingsStore@1",
              "target": [Circular],
              "values": Map {
                "settings" => Object {
                  "native_lang": "",
                  "translated_lang": "",
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "SettingsStore@1.settings.keys",
                      "observers": Set {},
                    },
                    "name": "SettingsStore@1.settings",
                    "pendingKeys": Map {
                      "cheerio" => false,
                      "$$typeof" => false,
                      "nodeType" => false,
                      "@@__IMMUTABLE_ITERABLE__@@" => false,
                      "@@__IMMUTABLE_RECORD__@@" => false,
                      "_isMockFunction" => false,
                      "toJSON" => false,
                    },
                    "proxy": [Circular],
                    "target": Object {
                      "native_lang": "",
                      "translated_lang": "",
                      Symbol(mobx administration): [Circular],
                    },
                    "values": Map {
                      "native_lang" => "",
                      "translated_lang" => "",
                    },
                  },
                },
              },
            },
          },
          "translationCategoryStore": TranslationCategoryStore {
            "create": [Function],
            "list": [Function],
            "rootStore": [Circular],
            "service": TranslationCategoryService {},
            "translationCategories": Array [
              Object {
                "id": 1,
                "title": "Some category",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                    "observers": Set {},
                  },
                  "name": "TranslationCategoryStore@2.translationCategories[..]",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "id" => 1,
                    "title" => "Some category",
                  },
                },
              },
            ],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "TranslationCategoryStore@2.keys",
                "observers": Set {},
              },
              "name": "TranslationCategoryStore@2",
              "target": [Circular],
              "values": Map {
                "translationCategories" => Array [
                  Object {
                    "id": 1,
                    "title": "Some category",
                    Symbol(mobx administration): ObservableObjectAdministration$$1 {
                      "defaultEnhancer": [Function],
                      "keysAtom": Atom$$1 {
                        "diffValue": 0,
                        "isBeingObserved": false,
                        "isPendingUnobservation": false,
                        "lastAccessedBy": 0,
                        "lowestObserverState": 2,
                        "name": "TranslationCategoryStore@2.translationCategories[..].keys",
                        "observers": Set {},
                      },
                      "name": "TranslationCategoryStore@2.translationCategories[..]",
                      "pendingKeys": Map {
                        "cheerio" => false,
                        "$$typeof" => false,
                        "nodeType" => false,
                        "@@__IMMUTABLE_ITERABLE__@@" => false,
                        "@@__IMMUTABLE_RECORD__@@" => false,
                        "_isMockFunction" => false,
                        "toJSON" => false,
                      },
                      "proxy": [Circular],
                      "target": Object {
                        "id": 1,
                        "title": "Some category",
                        Symbol(mobx administration): [Circular],
                      },
                      "values": Map {
                        "id" => 1,
                        "title" => "Some category",
                      },
                    },
                  },
                ],
              },
            },
          },
          "translationStore": [Circular],
        },
        "service": TranslationService {},
        "translations": Array [],
        "update": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "TranslationStore@3.keys",
            "observers": Set {},
          },
          "name": "TranslationStore@3",
          "target": [Circular],
          "values": Map {
            "translations" => Array [],
          },
        },
      }
    }
  />
</main>
`;
